### PR TITLE
remove /k8s.io/kubernetes/pkg/kubectl/testing

### LINKED ISSUE #164: Understand and potentially remove `k8s.io/kubernetes/pkg/kubectl/testing`
I don't understand what this is for. The `doc.go` file is plain wrong as it's just been copy-pasted from somewhere else (likely `k8s.io/kubernetes/pkg/kubectl`) and left unmodified.

Also the code is barely used anywhere, I'm wondering if we could remove that.

### PR DESCRIPTION
**Pull request by Lion-Wei**
**What this PR does / why we need it**:

The package `/k8s.io/kubernetes/pkg/kubectl/testing` only have one type `TestStruct` and only used in `printers_test.go`, so removed it and created a similar type in `printers_test.go`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubectl#164

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by cblecker** [comment]
```
W1208 19:08:22.522] Some packages in hack/.golint_failures do not exist anymore. Please remove them.
W1208 19:08:22.522] 
W1208 19:08:22.522]   pkg/kubectl/testing
```

**Comment by Lion-Wei** [comment]
@cblecker Thanks.

File: pkg/printers/internalversion/printers_test.go [hunk 1]
```diff
@@ -50,30 +50,66 @@ import (
...
+type TestStruct struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Key               string         `json:"Key"`
+	Map               map[string]int `json:"Map"`
+	StringList        []string       `json:"StringList"`
+	IntList           []int          `json:"IntList"`
+}
+
+func (in *TestStruct) DeepCopyObject() runtime.Object {
```
Comment (sttts) [inline pkg/printers/internalversion/printers_test.go]: I don't like this. The code is autogenerated. Generation does not work for _test files. That's why we had the testing dir.

**Comment by sttts** [comment]
We loose the autogeneration, i.e. increaes the maintenance burden with this PR. Please leave the files in the testing dir for this reason.

**Comment by apelisse** [comment]
> We loose the autogeneration, i.e. increaes the maintenance burden with this PR.

Agreed. Do we have to use this specific struct though? Or can we use another one? This package feels kind of out of place.

**Comment by Lion-Wei** [comment]
Seems like`DeepCopyObject` is needed to realize that interface.

**Comment by sttts** [comment]
> Agreed. Do we have to use this specific struct though? Or can we use another one? This package feels kind of out of place.

What is the alternative? We could use any non-trivial type from k8s.io/api. Haven't looked into the test details.

**Comment by sttts** [comment]
> Seems likeDeepCopyObject is needed to realize that interface.

If you are lucky, you can replace the body with `panic("never called")`. That would be fine for me if it works. I just don't like duplicating code that was originally generated. It's not maintainable.

**Comment by Lion-Wei** [comment]
@sttts Thanks for your advise, I have studied the detail. We do need `DeepCopyObject` in `TestVersionedPrinter`, but can use `TestPrintType` instead, and use `TestStruct` only for test print result.
Besides, maybe we can use a better name, I think. : )

File: pkg/printers/internalversion/printers_test.go [hunk 2]
```diff
@@ -50,30 +50,42 @@ import (
...
+	panic("never called")
```
Comment (sttts) [inline pkg/printers/internalversion/printers_test.go]: 👍

**Comment by Lion-Wei** [comment]
@deads2k @sttts can you help me add a LGTM, thanks.

**Comment by Lion-Wei** [comment]
@sttts Thanks. : )

**Pull request merged**